### ISSUE TITLE
Web component: Image with caption - issues related to modal

### ISSUE THREAD

**Issue by Praveen-sr** [carbon-design-system/carbon-for-ibm-dotcom#4266]
## Detailed description
On Voiceover iPhone
Issue: 
1. after activating the image alt text button, for the screen reader users the focus  remains on the triggering control itself
2. After activating close button,  the dialog got closed visually and the focus was taken to the parent page.
But screen reader users was not able to navigate using left and right swipe screen reader was only announcing as "close button"

Expected: 
1. Set the focus to the close button.
2. after activation screen reader users should be able to use left and right swipe.
